internal/controller/controllers/cluster_controller_test.go

```diff
@@ -0,0 +1,72 @@
+package controllers
+
+import (
+	"context"
+	"github.com/openshift/assisted-service/internal/controller/api/v1alpha1"
+	"github.com/stretchr/testify/assert"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/kubernetes/scheme"
+	ctrl "sigs.k8s.io/controller-runtime"
+	fakeclient "sigs.k8s.io/controller-runtime/pkg/client/fake"
+	"testing"
+)
+
+const (
+	kind       = "Cluster"
+	apiVersion = "adi.io.my.domain/v1alpha1"
+	namespace  = "test-namespace"
+)
+
+func newTestClusterReconciler(initObjs ...runtime.Object) *ClusterReconciler {
+	return &ClusterReconciler{
+		Client: fakeclient.NewFakeClient(initObjs...),
+		Scheme: scheme.Scheme,
+		Log:    getTestLog(),
+	}
+}
+
+func newTestDefaultCluster(t *testing.T) *v1alpha1.Cluster {
+	return &v1alpha1.Cluster{
+		Spec: v1alpha1.ClusterSpec{},
+		TypeMeta: metav1.TypeMeta{
+			Kind:       kind,
+			APIVersion: apiVersion,
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      t.Name(),
+			Namespace: namespace,
+		},
+	}
+}
+
+func newRequest(c *v1alpha1.Cluster) ctrl.Request {
+	return ctrl.Request{
+		NamespacedName: types.NamespacedName{
+			Name:      c.ObjectMeta.Name,
+			Namespace: c.ObjectMeta.Namespace,
+		},
+	}
+}
+
+func tryClusterReconcile(cr *ClusterReconciler, cluster *v1alpha1.Cluster) (ctrl.Result, error) {
+	req := newRequest(cluster)
+	res, err := cr.Reconcile(req)
+	if err != nil {
+		return res, err
+	}
+	modifiedCluster := &v1alpha1.Cluster{}
+	cr.Get(context.TODO(), req.NamespacedName, modifiedCluster)
+	modifiedCluster.DeepCopyInto(cluster)
+	return res, nil
+}
+
+func TestClusterReconciler_Reconcile(t *testing.T) {
```

Comment:
you are mixing between the default testing package and `ginkgo`
`ginkgo` test are under `Describe` take a look at every other test we have. 
golang testing tools identify tests by names that start with `func Test...`